lib/orm.php

```diff
@@ -2541,4 +2594,133 @@ public function __unset( $key ) {
 	public function __isset( $key ) {
 		return $this->offsetExists( $key );
 	}
+
+	/**
+	 * Whether this site has persistent caching implemented.
+	 *
+	 * @return bool
+	 */
+	private static function has_persistent_cache() {
```

Comment:
In general, I would like us to explore the possibility of depending only on the `wp_using_ext_object_cache()` method for assessing whether there is persistent cache and not on the existence of files. A site might as well have a `WP_CONTENT_DIR . '/object-cache.php'` file, without it meaning that persistent object cache is active there.

It would be awesome if we had something like this for the `has_persistent_cache()` method (this also adds filters to hard enable/disable our caching):
```
	private static function has_persistent_cache() {
		if ( self::$has_persistent_cache === false || self::$has_persistent_cache === true ) {
			return self::$has_persistent_cache;
		}

		 /**
		 * Filter: 'wpseo_force_enable_indexable_cache' - Allows force enabling indexable caching.
		 *
		 * @api bool Whether indexable cache is forced to be enabled.
		 */
		if ( \apply_filters( 'wpseo_force_enable_indexable_cache', false ) ) {
			self::$has_persistent_cache = true;
			return true;
		}

		// Check if the site is using an external object cache.
		if ( \function_exists( 'wp_using_ext_object_cache' ) ) {
			/**
			 * Filter: 'wpseo_force_disable_indexable_cache' - Allows force disabling indexable caching.
			 *
			 * @api bool Whether indexable cache is forced to be disabled.
			 */
			if ( \apply_filters( 'wpseo_force_disable_indexable_cache', false ) ) {
				self::$has_persistent_cache = false;
				return false;
			}

			self::$has_persistent_cache = true;
			return true;
		}

		// No persistent caching was detected.
		self::$has_persistent_cache = false;
		return false;
	}
```
but I'm worried about the 
```
		// This _should_ be covered by the `wp_using_ext_object_cache()` check,
		// but sometimes isn't depending on the timing of the cache.
```
remark you've added @aristath 🤔 Can we talk about this?

Reply:
I feel like `wp_using_ext_object_cache` should be enough here. The rest seems like overkill.

Reply:
Awesome, thanks for your input! We're probably gonna rely solely on that then. 

(We do want to keep the filters proposed above, for allowing users to hard enable/disable the enhancement, for taking care of edge cases)

Reply:
Why do you care that the object cache in persistent? WordPress fallbacks to an in memory cache if persistent cache is not present. But even this may mean saving some database queries.

Reply:
Because the plugin already employs a custom cache implementation (via `Meta_Tags_Context_Memoizer`) that covers a lot of the benefits of a memory cache, within single page loads.

There's the possibility of enabling this for every setup regardless of object cache being persistent or not (thus providing performance benefits for all Yoast websites), but this should be staggered among multiple releases imo, for risk management reasons amongst others.